### ISSUE TITLE
Regression with signing plugin using 'useGpgCmd()'

### ISSUE THREAD

**Issue by adude3141** [gradle/gradle#8657]
After upgrading from gradle 4.10.3 to gradle 5.2.1 we are not able to sign artefacts as before.

As [documentation](https://docs.gradle.org/5.2.1/userguide/signing_plugin.html) states

> Without any further configuration the gpg2 (on Windows: gpg2.exe) executable found on the PATH will be used. The password is supplied by the gpg-agent and the default key is used for signing.

### Expected Behavior
We would expect that if no signing key is specified, the signing plugin would fall back to gpg's default key.

### Current Behavior
With gradle 4 a 
```
./gradlew uploadArchives
```
sign and installs the artefacts in configured repository.
With gradle 5.1+ `uploadArchives` fails with
```
$ ./gradlew uploadArchives
> Task :signArchives FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':signArchives'.
> No value has been specified for property 'signatory.keyId'.
```

### Context
Unfortunately the error message is misleading as it complains about missing property 'signatory.keyId'. To get this running with gradle 5.1+ we need to specify
```
./gradlew uploadArchives -Psigning.gnupg.keyName=<KEY-ID>
```

With gradle4 there was no need to specify the signing key id.

### Steps to Reproduce (for bugs)
Run [attached sample project](https://github.com/gradle/gradle/files/2915783/signingGradle.zip) with either gradle 4 (which works) or gradle 5.1+ (which fails)

**Comment by crazybob** [comment]
This bit me, too. Thank you for providing a workaround.

**Comment by ljacomet** [comment]
With `uploadArchives` now deprecated and replaced by the publish plugins, `maven-publish` and `ivy-publish`, this will not be fixed.

**Comment by gibahjoe** [comment]
I just encountered this error also in `signArchives` task. Exactly what OP described and the workaround was exactly the same. `signArchives` isn't deprecated is it?

**Comment by zx2c4** [comment]
I ran into this error as well, not doing anything fancy at all, just using the normal signing plugin. Can this be reopened, @ljacomet?

**Comment by rschmitt** [comment]
I (continue to) have the same complaint as @zx2c4. Am I doing something weird in [my build](https://github.com/rschmitt/dynamic-object/blob/b334599369d5aab32fdd5a455053404058a7f23a/build.gradle.kts) that would cause this?

**Comment by osipxd** [comment]
@ljacomet, this issue still reproducible on Gradle 6.8.2 with `maven-publish` plugin

**Comment by micheljung** [comment]
@ljacomet this isn't specific to `uploadArchives`, happens with `gradle sign`, too.

**Comment by dengshiwei** [comment]
> @ljacomet this isn't specific to `uploadArchives`, happens with `gradle sign`, too.
how to solve it?

**Comment by ljacomet** [comment]
Thanks for following up on this.
Can someone share a reproducer?

Based on it, that issue could be re-opened indeed.

**Comment by micheljung** [comment]
@ljacomet a reproducer is available in #16597

**Comment by ljacomet** [comment]
Great, thanks for linking both issues. This one can stay closed then.
